packages/StepTracker/Step/Step.jsx

```diff
@@ -1,30 +1,35 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import classNames from 'classnames'
+
+import Icon from '@tds/core-decorative-icon'
+
+import styles from '../StepTracker.modules.scss'
 
 /**
  * A single step of a StepTracker.
- *
- * _This component can only be used as a child of `StepTracker`, and must be accessed as a
- * name-spaced component: `StepTracker.Step`._
- *
- * @see See [StepTracker](#steptracker) for usage.
  */
-const Step = ({ label, status, stepNumber }) => {
-  const cls = classNames('step-tracker__step', {
-    'step-tracker__step--completed': status === 'completed',
-    'step-tracker__step--processing': status === 'processing',
-    'step-tracker__step--waiting': status === 'waiting',
-  })
+
+const Step = ({ label, status, stepNumber, stepIndex }) => {
+  const cls = [styles.step]
+  if (status > stepIndex) {
+    cls.push(styles.completed)
+  } else if (status === stepIndex) {
+    cls.push(styles.processing)
+  }
+
   /* eslint-disable jsx-a11y/aria-props */
   return (
     <li
-      className={cls}
+      className={cls.join(' ')}
       aria-label={label}
-      aria-current={status === 'processing' ? 'true' : 'false'}
+      aria-current={status === stepIndex ? 'true' : 'false'}
     >
-      <span className="step-tracker__icon" />
-      <span className="step-tracker__label">
+      <span className={styles.icon}>
+        <span style={{ opacity: status > stepIndex ? 1 : 0 }}>
```

Comment:
I am thinking this opacity is effectively hiding and showing the icon? Is it perhaps better to just not render the Icon instead? If the circle has a set size, then the presence of the Icon shouldn't change that right?

If you do need to do this, perhaps you could use the shared `Display.modules.scss` classes that we have instead. You could use the `invisible` class to set the visibility rather than using the inline styles. That class name may make the intention here more apparent.

Reply:
Yeah, it's meant to hide it. I did it because there's less chance of performance loss or jank if we're not putting things in and out of the DOM. That class sounds like a good idea though, so I'll use that.

Reply:
Trying it again, another issue with pulling it from the DOM selectively results in the step label text shifting upward. I also noticed that the `Icon` component needs to have `visibility: inherit` applied to it for visibility to work. Might be worth looking over the shared `Icon` styling at some point in the future. I guess since we've established we want to do things the React way, I'll try to find a workaround or fix for this.

Reply:
(Figured out a workaround FYI, we should be good now.)